### PR TITLE
Turn the library from single-config to being instanced

### PR DESCRIPTION
**Pull request by robotmay**
This allows multiple configurations to be present in the same application.
This is a breaking change version with no backwards compatibility.

### CODE REVIEW

**Comment by robotmay** [comment]
This will break anything using it currently! Will release it as a new major version of the gem. Can you check this is vaguely sensible? :D

File: lib/codelocks/client.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    def access_key
+      @access_key || ENV['CODELOCKS_ACCESS_KEY']
+    end
+
+    # Faraday connection object
+    #
+    # @return [Faraday]
+
+    def connection
+      @connection ||= Faraday.new(url: base_uri) do |faraday|
```
Comment (philipcunningham) [inline lib/codelocks/client.rb:35]: See review comment.

**Review by philipcunningham — APPROVED** [review]
This looks really good.

See note in code: do you happen to know whether or not the `#connection` method returns an object that can be shared across multiple threads or should a new one be instantiated every time?

**Comment by robotmay** [comment]
Er, good point. I actually haven't tested how threadsafe it is :sweat_smile:
I think Faraday is but I'm not sure how threadsafe my own code is here. I'll test it out in the PS app I think, as that should show any problems fairly quickly.

**Pull request closed without merging**